docs/tutorial.md

```diff
@@ -737,3 +731,90 @@ Note that even though adding type annotations to .lua files makes it invalid
 Lua, you can still do so and load them from the Lua VM once the Teal package
 loader is installed by calling tl.loader().
 
+
+### Type definitions for third party libraries
+
+You can also create declaration files to annotate the types of third-party Lua
+modules, including C Lua modules. To do so, create a file with the .d.tl
+extension and require it as normal, i.e. `require("minetest")`.
+
+Types defined in this module will will be used as a source of type information 
+checking with `tl check`, even though the real Lua module will be loaded
+instead when requiring the module from Lua or `tl run`.
+
+If the third party library is "globally available", i.e. you do not explicitly
+`require` it, you can inject the types without affecting your generated code
+with a `tlconfig.lua`:
+
+```lua
+return {
+    preload_modules = {
```

Comment:
(nitpick: all code examples in this doc are indented with 3 spaces)

Reply:
As in - the code examples *should* be indented with 3 spaces?

Reply:
yup, to keep visual consistency with the ones that are already there

Reply:
Gotcha, I thought I had copy/pasted wrong when I saw 3-space indents. 😬